### PR TITLE
Add micrvom state serialization benchmark report.

### PR DESCRIPTION
**Pull request by sandreim**
## Reason for This PR

This PR is publishing basic synthetic state serialization benchmarks. As snapshot functionality becomes available we will publish the a full breakdown of microvm save/restore for various configurations:
- state serialization/deserialization
- kvm state save/restore
- device save/restore
- overall save/restore latency.

## Description of Changes

In commit messages.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.

### CODE REVIEW

File: docs/benchmarks/state-serialize.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
-### Snapshot save/load benchmarks
-Snapshot contains 100 structs and a 10k array.  
+
+### MicroVM state serializiation benchmarks
```
Comment (lauralt) [inline docs/benchmarks/state-serialize.md:2]:
```suggestion
### MicroVM state serialization benchmarks
```

File: docs/benchmarks/state-serialize.md [hunk 2]
```diff
@@ -29,7 +30,12 @@ Snapshot size: 83886 bytes.
...
-Deserialize + crc64: **248.62 us**
+
+| Test                |      Mean     |
+|---------------------|---------------|
+| Serialize           |    354.18 us  |
+| Serialize + crc64   |    406.08 us  |
+| Deserialize         |    61.412 us  |
+| Deserialize + crc64 |    258.27 us  |
+
+Detailed criterion benchmarks available [here](https://s3.amazonaws.com/spec.ccfc.min/perf/snapshot-0.23/report/index.html)
```
Comment (lauralt) [inline docs/benchmarks/state-serialize.md:41]: nit: full stop.

**Comment by sandreim** [comment]
> nit: I think the commits could be squashed since the new results are obtained I suppose for the new sample size.

One changes code, the other changes docs. I think they need to be reviewed separately.

File: docs/benchmarks/state-serialize.md (hunk 2 above)
Comment (sandreim) [reply docs/benchmarks/state-serialize.md]: done

File: docs/benchmarks/state-serialize.md (hunk 1 above)
Comment (sandreim) [reply docs/benchmarks/state-serialize.md]: done

**Comment by sandreim** [comment]
I've adressed the comments, please review again @aghecenco @lauralt

**Comment by lauralt** [comment]
> > nit: I think the commits could be squashed since the new results are obtained I suppose for the new sample size.
> 
> One changes code, the other changes docs. I think they need to be reviewed separately.

Then that code change should be somehow documented (i.e. explain a bit in the commit message why you did the sample size change, even if for us it may seem obvious).

**Pull request merged**